### ISSUE TITLE
Alsaaudio missing from dependencies

### ISSUE THREAD

**Issue by sambnh** [Dustpancake/Asteroid#31]
Upon running the tests, the following error occurred multiple times:
`ModuleNotFoundError: No module named 'alsaaudio'`
I'm assuming this is due to the alsaaudio module not being in the requirements.txt?

**Comment by Dustpancake** [comment]
alssaudio is explicitly for Linux OS - you can't install it on Windows or other *nix systems. If you check in `run.py` you'll see there's a basic mock of the module for when we run the player, providing the interface for what we need.

The reason this fails with the unit tests is because pytest loads and scans all of the .py files for the test structure - the fix is as simple as mocking the module in a `pytest.fixture` inserted before any test module is loaded. 

This is part of the larger problem the refactoring and config parser has introduced - the unit tests are out of date, and weren't 'proper' unit tests in the first place.

**Comment by Dustpancake** [comment]
```
(venv) Morgana:Asteroid lucifer$ pytest -v
===================================================================================== test session starts ======================================================================================
platform darwin -- Python 3.7.3, pytest-4.6.1, py-1.8.0, pluggy-0.12.0 -- /Users/lucifer/PycharmProjects/Asteroid/venv/bin/python3
cachedir: .pytest_cache
rootdir: /Users/lucifer/PycharmProjects/Asteroid
collected 48 items                                                                                                                                                                             

src/test/test_flaskserv/test_FlaskServer.py::TestFlaskServ::test_connection PASSED                                                                                                       [  2%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerUsers::test_add_user PASSED                                                                                                       [  4%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerUsers::test_get_user_by_id PASSED                                                                                                 [  6%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerUsers::test_get_all_users PASSED                                                                                                  [  8%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerMusic::test_get_songs PASSED                                                                                                      [ 10%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerMusic::test_get_song_by_id PASSED                                                                                                 [ 12%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerMusic::test_get_by_name PASSED                                                                                                    [ 14%]
src/test/test_flaskserv/test_FlaskServer.py::TestServerMusic::test_get_by_artist PASSED                                                                                                  [ 16%]
src/test/test_flaskserv/test_FlaskServer.py::TestPlaylist::test_vote_new PASSED                                                                                                          [ 18%]
src/test/test_flaskserv/test_FlaskServer.py::TestPlaylist::test_vote_update PASSED                                                                                                       [ 20%]
src/test/test_flaskserv/test_FlaskServer.py::TestPlaylist::test_vote_fail PASSED                                                                                                         [ 22%]
src/test/test_flaskserv/test_FlaskServer.py::TestPlaylist::test_fetch_playlist PASSED                                                                                                    [ 25%]
src/test/test_flaskserv/test_playqueue.py::TestPlaylist::test_add PASSED                                                                                                                 [ 27%]
src/test/test_flaskserv/test_playqueue.py::TestPlaylist::test_get_playlist PASSED                                                                                                        [ 29%]
src/test/test_flaskserv/test_playqueue.py::TestPlaylist::test_update_vote PASSED                                                                                                         [ 31%]
src/test/test_flaskserv/test_playqueue.py::TestPlaylist::test_get_most_voted PASSED                                                                                                      [ 33%]
src/test/test_flaskserv/test_playqueue.py::TestPlaylist::test_remove PASSED                                                                                                              [ 35%]
src/test/test_flaskserv/test_databases/test_Database.py::test_enter_exit PASSED                                                                                                          [ 37%]
src/test/test_flaskserv/test_databases/test_Database.py::test_create_table PASSED                                                                                                        [ 39%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_insert_row PASSED                                                                                            [ 41%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_select_rows PASSED                                                                                           [ 43%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_select_modifiers PASSED                                                                                      [ 45%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_get_count PASSED                                                                                             [ 47%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_latest_items PASSED                                                                                          [ 50%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_update PASSED                                                                                                [ 52%]
src/test/test_flaskserv/test_databases/test_Database.py::TestDatabase::test_delete PASSED                                                                                                [ 54%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::test_init PASSED                                                                                                                 [ 56%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::test_table_creation PASSED                                                                                                       [ 58%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::TestMDB::test_adding_song PASSED                                                                                                 [ 60%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::TestMDB::test_get_by_id PASSED                                                                                                   [ 62%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::TestMDB::test_get_by_name PASSED                                                                                                 [ 64%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::TestMDB::test_get_by_artist PASSED                                                                                               [ 66%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::TestMDB::test_get_page PASSED                                                                                                    [ 68%]
src/test/test_flaskserv/test_databases/test_MusicDB.py::TestMDB::test_unique PASSED                                                                                                      [ 70%]
src/test/test_flaskserv/test_databases/test_UserDB.py::test_init PASSED                                                                                                                  [ 72%]
src/test/test_flaskserv/test_databases/test_UserDB.py::test_table_creation PASSED                                                                                                        [ 75%]
src/test/test_flaskserv/test_databases/test_UserDB.py::TestUDB::test_add_user PASSED                                                                                                     [ 77%]
src/test/test_flaskserv/test_databases/test_UserDB.py::TestUDB::test_unique PASSED                                                                                                       [ 79%]
src/test/test_flaskserv/test_databases/test_UserDB.py::TestUDB::test_get_by_id PASSED                                                                                                    [ 81%]
src/test/test_player/test_ClientThread.py::test_run PASSED                                                                                                                               [ 83%]
src/test/test_player/test_ClientThread.py::test_handle PASSED                                                                                                                            [ 85%]
src/test/test_player/test_ConditionObject.py::TestConditionObject::test_init_state PASSED                                                                                                [ 87%]
src/test/test_player/test_ConditionObject.py::TestConditionObject::test_lock_exists PASSED                                                                                               [ 89%]
src/test/test_player/test_INETServer.py::test_startup PASSED                                                                                                                             [ 91%]
src/test/test_player/test_INETServer.py::test_attach PASSED                                                                                                                              [ 93%]
src/test/test_player/test_INETServer.py::test_start PASSED                                                                                                                               [ 95%]
src/test/test_player/test_PlayStream.py::test_restrict_call PASSED                                                                                                                       [ 97%]
src/test/test_player/test_PlayStream.py::TestPlayStream::test_play PASSED                                                                                                                [100%]

======================================================================================= warnings summary =======================================================================================
venv/lib/python3.7/site-packages/jinja2/utils.py:485
  /Users/lucifer/PycharmProjects/Asteroid/venv/lib/python3.7/site-packages/jinja2/utils.py:485: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import MutableMapping

venv/lib/python3.7/site-packages/jinja2/runtime.py:318
  /Users/lucifer/PycharmProjects/Asteroid/venv/lib/python3.7/site-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

src/test/test_player/test_PlayStream.py::TestPlayStream::test_play
  /Users/lucifer/PycharmProjects/Asteroid/src/main/player/PlayStream.py:99: DeprecationWarning: The binary mode of fromstring is deprecated, as it behaves surprisingly on unicode inputs. Use frombuffer instead
    val = np.fromstring(data, dtype=np.short)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================================================ 48 passed, 3 warnings in 0.38 seconds =============================================================================

```

Fixed in e2ec65bcbaf67f6dadca501745fa0de9653106ec.

**Comment by Dustpancake** [comment]
For future note, we moved to PyAudio -- installation notes are [here](https://github.com/Dustpancake/Dust-Notes/blob/master/raspi/pyaudio.md?fbclid=IwAR0HfwzsSe0gLFFAS5blM9lqNu45stcIKGPJlNViXopWEy0H-xtRwVFoEjA) for pyaudio on raspi.
